docs/specifications/tests/Basic-TP/basic00.md

```diff
@@ -1,63 +1,314 @@
+B00_INITIAL_DOT          | CRITICAL | Domain name starts with dot.
+B00_REPEATED_DOTS        | CRITICAL | Domain name has repeated dots.
+B00_INVALID_ASCII        | CRITICAL | ASCII label has characters not permitted.
+B00_INVALID_U_LABEL      | CRITICAL | Label with non-ASCII character is not valid U-label.
+B00_LABEL_TOO_LONG       | CRITICAL | Domain name has a label that is too long (more than 63 characters).
+B00_DOMAIN_NAME_TOO_LONG | CRITICAL | Domain name is too long (more than 253 characters with no final dot).
+
+
+## Ordered description of steps to be taken to execute the test case
+
+1.  Create an empty, ordered list of labels ("Child Labels").
+
+2.  Remove any and all initial and final space (U+0020) from *Child Zone*.
+
+3.  If *Child Zone* is not encoded in UTF-8 (ASCII is a subset of UTF-8) then
+    output *[B00_NOT_UTF-8_ASCII]* and terminate this test case.
+
+4.  Replace all instances of dots from *Full Stops* in *Child Zone* with U+002E
+    (dot ".").
+
+5.  If *Child Zone* starts with dot (".") then output *[B00_INITIAL_DOT]* and
+    terminate this test.
+
+6.  If *Child Zone* has any instance of two or more consecutive dots (".") then
+    output *[B00_REPEATED_DOTS]* and terminate this test.
+
+7.  Remove trailing dot (".") from *Child Zone* unless the length is 1 (i.e. root
+    zone name).
+    
+8.  If *Child Zone* os the root zone (i.e. ".") then terminate this test case.
+
+9.  Split *Child Zone* into labels by dot "." and put them in the same order in 
+    *Child Labels*.
+
+10. For each "Label" in *Child Labels* do:
+    1. If all characters in *Label* are ASCII characters, then do:
+       1. If any character in *Label* is not listed in *Valid ASCII*, then output
+          [B00_INVALID_ASCII] and *Label*, and terminate this test case.
+       2. Else, make all characters in *Label* lower case (a-z) if tagged as
+          upper case (A-Z) according to *Unicode* and update the label in
+          *Child Labels*. Mind U+0049 (LATIN CAPITAL LETTER I) in Turkish and
+          Azeri locale.
+    2. Else do:
+       1. Assume that *Label* is a U-label.
+       2. Make all characters in *Label* lower case if tagged as upper case
+          according to *Unicode* and update the label in *Child Labels*. Mind
+          U+0049 (LATIN CAPITAL LETTER I) in Turkish and Azeri locale.
+       3. Try to convert *Label* to an A-label according to the IDNA2008
+          algorithm (not IDNA2003).
+          1. If the conversion failed, then output *[B00_INVALID_U_LABEL]*
+             and *Label*, and terminate this test case.
+          2. Else, replace the U-label in *Child Labels* with the A-label from
+             the conversion.
+    3. Go to next label.
+
+11. For each "Label" in *Child Labels* do:
+    1. If the length (number of character) in *Label* is greater than 63 then
+       output *[B00_LABEL_TOO_LONG]* and the label, and terminate this test case.
```

Comment:
> `*[B00_LABEL_TOO_LONG]* and the label`

maybe use the same wording as above with `and *Label*`

Reply:
Fixed.